x/farming/keeper/proposal_handler.go

```diff
@@ -103,8 +105,13 @@ func (k Keeper) UpdatePublicPlanProposal(ctx sdk.Context, proposals []*types.Upd
 			return sdkerrors.Wrapf(sdkerrors.ErrNotFound, "plan %d is not found", p.GetPlanId())
 		}
 
-		switch plan := plan.(type) {
-		case *types.FixedAmountPlan:
+		if p.EpochAmount.IsAllPositive() {
```

Comment:
We should check if both `EpochAmount` and `EpochRatio` are set at the same time. If both of those fields were set, then it should reject the proposal to disambiguate.

Reply:
I think that is being handled in [ValidateBasic](https://github.com/tendermint/farming/blob/master/x/farming/types/proposal.go#L47) when a user sends a governance proposal, isn't it?

Reply:
Oh, ok. You're correct.